classifiers/dns.go

```diff
@@ -0,0 +1,30 @@
+package classifiers
+
+import (
+	"github.com/google/gopacket/layers"
+	"github.com/mushorg/go-dpi"
+)
+
+type DnsClassifier struct{}
+
+func (_ DnsClassifier) HeuristicClassify(flow *godpi.Flow) bool {
+	if len(flow.Packets) == 0 {
+		return false
+	}
+	for _, packet := range flow.Packets {
+		if layer := (*packet).Layer(layers.LayerTypeUDP); layer != nil {
+			srcPort := layer.(*layers.UDP).SrcPort
+			dstPort := layer.(*layers.UDP).DstPort
+			if srcPort != 53 && dstPort != 53 {
```

Comment:
I'm returning false if there is a packet in the flow for which (src != 53 && dst != 53). That's equivalent to !(src == 53 || dst == 53). If I did  (src != 53 || dst != 53) I would return false for all packets except those that are both from and to port 53.